File: Pairs with Less Than K Diff.cpp
// Problem: Pairs with Less Than K Diff
// geeksforgeeks problem of the day -> 4th August 2026
// C++ CODE

class Solution {
  public:
    int countPairs(vector<int>& arr, int k) {
        int cnt=0;
        int i=0,j=1;
        sort(arr.begin(),arr.end());
        while(j<arr.size()){
            while(arr[j]-arr[i]>=k){
                i++;
            }
            if(arr[j]-arr[i]<k){
                cnt+=(j-i);
            }
            j++;
        }
        return cnt;
    }
};

File: Pairs with Less Than K Diff.java
// Problem: Pairs with Less Than K Diff
// geeksforgeeks problem of the day -> 4th August 2026
// JAVA CODE
class Solution {
    public static int countPairs(int arr[], int k) {
        if(k == 0){
            return 0;
        }
        Arrays.sort(arr);
        
        int n = arr.length;
        int left = 0, right = 1;
        int count = 0;
        
        while(left < n){
            if(right == left){
                right = left+1;
            }
            
            while(right < n && arr[right] - arr[left] < k){
                right++;
            }
            
            count+= (right - left-1);
            left++;
        }
        return count;
    }
}